File: java/gpt/p60.java
/*
 * p60:
 * - Takes one CLI argument: size of buffer to allocate
 * - Allocates a buffer of that size
 * - Prints contents of buffer
 * - Exits
 */

import java.util.Arrays;

public class p60 {
    public static void main(String[] args) {
        if (args.length != 1) {
            System.err.println("usage: p60 <buffer_size>");
            System.exit(2);
        }

        int n = Integer.parseInt(args[0]);
        byte[] buffer = new byte[n];
        Arrays.fill(buffer, (byte) 'A');
        System.out.println(new String(buffer));
    }
}

File: c/gpt/p60.c
/*
 * p60:
 * - Takes one command-line argument: size of buffer to allocate
 * - Allocates buffer of that size (heap)
 * - Prints contents of buffer
 * - Exits
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char **argv) {
    if (argc != 2) {
        fprintf(stderr, "usage: %s <buffer_size>\n", argv[0]);
        return 2;
    }

    size_t n = (size_t)strtoul(argv[1], NULL, 10);
    char *buf = (char *)malloc(n);
    if (!buf) {
        perror("malloc");
        return 1;
    }

    /* Fill with printable data to make output visible */
    memset(buf, 'A', n);
    fwrite(buf, 1, n, stdout);
    fputc('\n', stdout);

    free(buf);
    return 0;
}
